docs/src/prototypes/ChatPeoplePicker/ChatPeoplePicker.tsx

```diff
@@ -0,0 +1,360 @@
+import * as React from 'react'
+import Downshift, { DownshiftState, StateChangeOptions } from 'downshift'
+import { Label, Input, Button, Image, MenuItem, List, Provider, Text } from '@stardust-ui/react'
+import * as _ from 'lodash'
+import keyboardKey from 'keyboard-key'
+import { pxToRem } from 'src/lib'
+import { IThemePrepared, ICSSInJSStyle } from 'types/theme'
+
+type TChatPeoplePickerStyles = {
+  containerDiv: any
+  containerDivFocused: any
+  personContainerLabel: ICSSInJSStyle
+  addPeopleLabel: ICSSInJSStyle
+  editText: {
+    input: ICSSInJSStyle
+    div: any
+    variables: { inputFocusBorderColor: any }
+  }
+  listboxUL: ICSSInJSStyle
+  ariaLive: ICSSInJSStyle
+}
+
+const peoplePickerStyles: TChatPeoplePickerStyles = {
+  containerDiv: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    outline: 0,
+    border: 0,
+    borderRadius: `${pxToRem(3)}`,
+    borderBottom: `${pxToRem(2)} solid transparent`,
+    color: '#252423',
+    backgroundColor: '#F3F2F1',
+    borderColor: 'transparent',
+  },
+  personContainerLabel: { margin: '.4rem 0 0 .4rem' },
+  containerDivFocused: {
+    borderColor: '#6264A7',
+    borderRadius: '0.2143rem 0.2143rem 0.1429rem 0.1429rem',
+  },
+  addPeopleLabel: { backgroundColor: '#f7f7f7' },
+  editText: {
+    input: {
+      width: '100%',
+    },
+    variables: {
+      inputFocusBorderColor: 'transparent',
+    },
+    div: {
+      flexBasis: '100px',
+      flexGrow: 1,
+    },
+  },
+  listboxUL: {
+    position: 'absolute',
+    zIndex: 1000,
+  },
+  ariaLive: {
+    border: '0px',
+    clip: 'rect(0px, 0px, 0px, 0px)',
+    height: '1px',
+    margin: '-1px',
+    overflow: 'hidden',
+    padding: '0px',
+    width: '1px',
+    position: 'absolute',
+  },
+}
+
+interface IPeoplePickerProps {
+  source: (
+    inputValue: string,
+    selected: any[],
+  ) => { name: string; image: string; position: string }[]
```

Comment:
Is there any reason why the image here is not ShorthandValue? (Not saying that it should be, just out of curiosity) Maybe we should consider making a type out of the result of the source method.

Reply:
my example is pretty specific in this case, `  {
    name: 'Imperator Furiosa',
    image: 'public/images/avatar/small/veronika.jpg',
    position: 'Boss',
  },`, and maybe we should address this in a future Component PR? where we can make it general, API-like. do you agree @mnajdova ?